orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/pipeline/BuildStage.groovy

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 Schibsted ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *   You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.igor.pipeline
+
+import com.netflix.spinnaker.orca.CancellableStage
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.batch.RestartableStage
+import com.netflix.spinnaker.orca.igor.tasks.MonitorBuildJobTask
+import com.netflix.spinnaker.orca.igor.tasks.MonitorQueuedBuildJobTask
+import com.netflix.spinnaker.orca.igor.tasks.StartBuildJobTask
+import com.netflix.spinnaker.orca.igor.tasks.StopBuildJobTask
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder
+import com.netflix.spinnaker.orca.pipeline.TaskNode
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.pipeline.model.Task
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+
+@Slf4j
+@CompileStatic
+abstract class BuildStage implements StageDefinitionBuilder, RestartableStage, CancellableStage {
+  @Autowired StopBuildJobTask stopBuildJobTask
+
+  @Override
+  <T extends Execution<T>> void taskGraph(Stage<T> stage, TaskNode.Builder builder) {
+    builder
+      .withTask("start${getType().capitalize()}Job", StartBuildJobTask.class)
+      .withTask("waitFor${getType().capitalize()}JobStart", MonitorQueuedBuildJobTask.class)
+
+    if (!stage.getContext().getOrDefault("waitForCompletion", "true").toString().equalsIgnoreCase("false")) {
+      builder.withTask("monitor${getType().capitalize()}Job", MonitorBuildJobTask.class);
+    }
+  }
+
+  @Override
+  Stage prepareStageForRestart(ExecutionRepository executionRepository, Stage stage, Collection<StageDefinitionBuilder> allStageBuilders) {
+    stage = StageDefinitionBuilder.StageDefinitionBuilderSupport
```

Comment:
is the travis stage the same as the jenkins stage in terms of buildInfo and buildNumber?